static/js/containers/ChannelPage.js

```diff
@@ -16,41 +16,68 @@ import { getChannelName } from "../lib/util"
 
 import type { Dispatch } from "redux"
 import type { Match } from "react-router"
+import type { RestState } from "../flow/restTypes"
+import type { Channel, Post } from "../flow/discussionTypes"
 
 class ChannelPage extends React.Component {
   props: {
     match: Match,
     dispatch: Dispatch,
-    channels: Object,
-    postsForChannel: Object,
-    posts: Object
+    channels: RestState<Map<string, Channel>>,
+    postsForChannel: RestState<Map<string, Array<string>>>,
+    posts: RestState<Map<string, Post>>,
+    subscriptions: RestState<Array<Channel>>,
   }
 
   componentWillMount() {
-    const { dispatch } = this.props
+    this.updateRequirements()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (getChannelName(this.props) !== getChannelName(prevProps)) {
+      // if the page was changed, fetch the new information for this channel
+      this.updateRequirements()
+    }
+  }
+
+  updateRequirements = () => {
+    const { dispatch, subscriptions } = this.props
     const channelName = getChannelName(this.props)
     dispatch(actions.channels.get(channelName))
     dispatch(actions.postsForChannel.get(channelName)).then(({ posts }) => {
       dispatch(setPostData(posts))
     })
+
+    if (!subscriptions.loaded && !subscriptions.processing) {
+      dispatch(actions.subscriptions.get())
+    }
   }
 
-  renderContents(channelName, channels, postsForChannel, posts) {
+  renderContents(channelName, channels, postsForChannel, posts, subscriptions) {
+    if (!channels.data || !posts.data || !subscriptions.data) {
+      return
+    }
     const channel = channels.data.get(channelName)
+    if (!postsForChannel.data) {
+      return
+    }
```

Comment:
It's a flow quirk, it doesn't understand that `get()` won't affect `postsForChannel`